src/compas_model/model/element_tree.py

```diff
@@ -34,3 +36,175 @@ def __init__(self, model=None, name="root", attributes=None):
+            return count
+
+        count = _count_elements(self.root, count)
+        return count
+
+    # ==========================================================================
+    # Printing
+    # ==========================================================================
+
+    def __repr__(self):
+        return "ElementTree with {} nodes".format(len(list(self.nodes)))
+
+    def __str__(self):
+        return self.__repr__()
+
+    def print(self):
+        """Print the tree in a readable format."""
+
+        def _print(node, depth=0):
+
+            # ------------------------------------------------------------------
+            # print current node
+            # ------------------------------------------------------------------
+            parent_name = "None" if node.parent is None else node.parent.name
+            print("-" * 100)
+            message = (
+                "    " * depth
+                + str(node)
+                + " "
+                + "| Parent: "
+                + parent_name
+                + " | Root: "
+                + node.tree.name
+            )
+
+            if depth == 0:
+                message += (
+                    " | Elements: " + "{" + str(node.tree.number_of_elements) + "}"
+                )
+
+            print(message)
+
+            # ------------------------------------------------------------------
+            # recursion
+            # ------------------------------------------------------------------
+            if node.children is not None:
+                for child in node.children:
+                    _print(child, depth + 1)
+
+        # ------------------------------------------------------------------
+        # main call for the recursive printing
+        # ------------------------------------------------------------------
+        _print(self.root)
+
+    # ==========================================================================
+    # Behavior - Tree
+    # ==========================================================================
+
+    def add_group(self, name=None, geometry=None, attributes=None, parent=None):
+        """Add a group node to the tree.
+
+        Parameters
+        ----------
+
+        name : str, optional
+            A name or identifier for the node.
+
+        geometry : Any, optional
+            Geometry or any other property, when you want to give a group a shape besides name.
+
+        attributes : dict, optional
+            A dictionary of additional attributes to be associated with the node.
+        """
+        return self.root.add_group(
+            name=name, geometry=geometry, attributes=attributes, parent=None
+        )
+
+    def add_element(self, name=None, element=None, attributes=None, parent=None):
+        """Add an element node to the tree.
+
+        Parameters
+        ----------
+        name : str, optional
+            A name or identifier for the node.
+
```

Comment:
there cannot be blank lines in parameter lists. they will not be processed correctly